### PR TITLE
[JENKINS-59247] - fix "Fail if path is not found"

### PR DESCRIPTION
**Pull request by MichaelSp**
fixes https://issues.jenkins-ci.org/browse/JENKINS-59247

Signed-off-by: Michael Sprauer <Michael.Sprauer@sap.com>

### CODE REVIEW

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 1]
```diff
@@ -54,6 +54,12 @@
 import hudson.model.TaskListener;
 import hudson.security.ACL;
 import hudson.tasks.BuildWrapperDescriptor;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: imports is A-Z and static last.

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 2]
```diff
@@ -125,8 +126,8 @@ public void setVaultAccessor(VaultAccessor vaultAccessor) {
         return leaseIds;
     }
 
-    private void provideEnvironmentVariablesFromVault(Context context, Run build,
-        EnvVars envVars) {
+    protected void provideEnvironmentVariablesFromVault(Context context, Run build,
+                                                        EnvVars envVars) {
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: Please revert formatting here

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 3]
```diff
@@ -143,7 +144,8 @@ private void provideEnvironmentVariablesFromVault(Context context, Run build,
                 .orElse(configuration.getEngineVersion());
             try {
                 LogicalResponse response = vaultAccessor.read(path, engineVersion);
-                parseVaultErrorCodes(path, response);
+                if (responseHasErrors(path, response))
+                    continue;
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: brackets or newline after continue

File: src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import com.datapipe.jenkins.vault.model.VaultSecret;
+import com.datapipe.jenkins.vault.model.VaultSecretValue;
+import hudson.EnvVars;
+import hudson.model.Build;
+import hudson.model.Run;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```
Comment (casz) [inline src/test/java/com/datapipe/jenkins/vault/VaultBuildWrapperTest.java]: import order A-Z, static last.

**Comment by casz** [comment]
Hi @MichaelSp Thanks for contribute would you mind taking a look at my review comments? :)

Hopefully we can land this :)

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 5]
```diff
@@ -96,34 +95,21 @@ public void setUp(Context context, Run<?, ?> build, FilePath workspace,
...
-        this.vaultAccessor = vaultAccessor;
-    }
-
-    private List<String> retrieveLeaseIds(List<LogicalResponse> logicalResponses) {
-        List<String> leaseIds = new ArrayList<>();
-        for (LogicalResponse response : logicalResponses) {
-            String leaseId = response.getLeaseId();
-            if (leaseId != null && !leaseId.isEmpty()) {
-                leaseIds.add(leaseId);
+    private void pullAndMergeConfiguration(Run<?, ?> build) {
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: please revert the sorting of methods 😰

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 6]
```diff
@@ -86,7 +85,7 @@ public VaultBuildWrapper(@CheckForNull List<VaultSecret> vaultSecrets) {
 
     @Override
     public void setUp(Context context, Run<?, ?> build, FilePath workspace,
-        Launcher launcher, TaskListener listener, EnvVars initialEnvironment) {
+                      Launcher launcher, TaskListener listener, EnvVars initialEnvironment) {
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: style config is located here: https://github.com/jenkinsci/hashicorp-vault-plugin/tree/master/.idea/codeStyles

Comment (MichaelSp) [reply src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: uuh, next line of parameters on the same indentation level as first code line of the methods 
but 🤷‍♂ its your code... 😅

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 7]
```diff
@@ -85,7 +85,7 @@ public VaultBuildWrapper(@CheckForNull List<VaultSecret> vaultSecrets) {
...
```
Comment (casz) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: Please revert indent 😓

**Comment by MichaelSp** [comment]
thanks

**Comment by casz** [comment]
Thanks for your contribution!

**Pull request merged**